lib/views/staging-view.js

```diff
@@ -377,20 +381,30 @@ export default class StagingView extends React.Component {
     this.subs.dispose();
   }
 
+  getSelectedItemFilePaths() {
+    return Array.from(this.state.selection.getSelectedItems(), item => item.filePath);
+  }
+
   getSelectedConflictPaths() {
     if (this.state.selection.getActiveListKey() !== 'conflicts') {
       return [];
     }
-    return Array.from(this.state.selection.getSelectedItems(), item => item.filePath);
+    return this.getSelectedItemFilePaths();
   }
 
   openFile() {
-    const filePaths = Array.from(this.state.selection.getSelectedItems(), item => item.filePath);
+    const filePaths = this.getSelectedItemFilePaths();
     return this.props.openFiles(filePaths);
   }
 
   discardChanges() {
-    const filePaths = Array.from(this.state.selection.getSelectedItems(), item => item.filePath);
+    const filePaths = this.getSelectedItemFilePaths();
+    addEvent('discard-unstaged-changes', {
```

Comment:
it looks like we're measuring similar events from the `StagingView` and the `FilePatchController`.  Since the `FilePatchController` is above the `StagingView` in the component tree, I'm assuming both these events will be fired every time a discard happens.  Which makes me worry about double counting, if you're trying to figure out the total usage of the "undo discard" feature.

What's really useful to learn from instrumenting the view layer is how the user is interacting with the UI.  Are they actually clicking the new button?  Instrumenting that button more specifically makes double counting less likely and clarifies what you're looking to learn.

Reply:
> Since the FilePatchController is above the StagingView in the component tree, I'm assuming both these events will be fired every time a discard happens.

They're actually in different parts of the component hierarchy and instrument separate discard/undo discard states. There should be no double counting.

> What's really useful to learn from instrumenting the view layer is how the user is interacting with the UI.

I totally agree with this. The tricky thing is the we can't easily instrument all UI actions to distinguish between different entry points. A button, yes, that's easy. Menu usage versus keybinding versus command palette usage, not so easy. I can't think of an easy straightforward way to instrument those, they all go through the command. 

In order to catch all discard and undo discard actions I had to instrument the `undoLastDiscard` method. I had the thought to also instrument the button but that would lead to duplicate events. I guess I could pass some meta data along to record if the event was triggered by a button versus command. But we'd have to keep in mind that the command could be triggered by a menu or keybinding as well as the command palette. Okay I'll do that... Maybe add an `eventSource` property.